### PR TITLE
fix: Use puppeteer for picuki

### LINKED ISSUE #11398: 🔴「Picuki」RSS 失效
### 路由地址

```routes
/picuki/profile/:id/:functionalFlag?
```

### 完整路由地址

```fullroutes
/picuki/profile/stefaniejoosten
```

### 相关文档

https://docs.rsshub.app/social-media.html#picuki-yong-hu

## 「官方演示」和「自建」均出现报错

失效了已经有一段时间了

https://rsshub.app/picuki/profile/stefaniejoosten

https://github.com/DIYgod/RSSHub/issues/11348

## Looks like something went wrong

Route requested: /profile/stefaniejoosten
Error message: Response code 403 (Forbidden): target website might be blocking our access, you can [host your own RSSHub instance](https://docs.rsshub.app/install/) for a better usability.
Helpful Information to provide when opening issue:
Path: /profile/stefaniejoosten
Node version: v16.18.1
Git Hash: 314b052

### 部署

RSSHub 演示 (https://rsshub.app)

### 部署相关信息

Node version: v16.18.1 Git Hash: 314b052

### 额外信息

```shell
Route requested: /profile/stefaniejoosten
Error message: Response code 403 (Forbidden): target website might be blocking our access, you can host your own RSSHub instance for a better usability.
Helpful Information to provide when opening issue:
Path: /profile/stefaniejoosten
Node version: v16.18.1
Git Hash: 314b052
```

### 这不是重复的 issue

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by devinryu**
## 该 PR 相关 Issue / Involved issue

Close #11398 

## 完整路由地址 / Example for the proposed route(s)

```routes
/picuki/profile/blackpinkofficial/1
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [x] 反爬/频率限制 anti-bot or rate limit?
  - [x] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [x] 添加了新的包 New package added
- [x] `Puppeteer`

### CODE REVIEW

File: lib/v2/picuki/utils.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        request.resourceType() === 'document' ? request.continue() : request.abort();
+    });
+    await page.goto(url, {
+        waitUntil: 'domcontentloaded',
+    });
+
+    await page.waitForSelector('.wrapper');
+
+    const html = await page.evaluate(() => document.documentElement.innerHTML);
+    return html;
```
Comment (TonyRL) [inline lib/v2/picuki/utils.js:15]: Please close the page after extracting the data
```js
await page.close();
```

File: lib/v2/picuki/profile.js [hunk 2]
```diff
@@ -31,12 +36,8 @@ module.exports = async (ctx) => {
...
     async function getMedia(url) {
         const getPost = async () => {
-            const response = await got(url, {
-                headers: {
-                    Referer: profileUrl,
-                },
-            });
-            return response.data;
+            const response = await puppeteerGet(url, browser);
+            return response;
```
Comment (TonyRL) [inline lib/v2/picuki/profile.js:40]: Please close the browser after all puppeteer page requests finish.
```js
await browser.close();
```

Comment (devinmugen) [reply lib/v2/picuki/profile.js:40]: Thank you for the reminder.

**Comment by yqlbu** [comment]
Looks like the issue are NOT resolved yet - https://rsshub.app/picuki/profile/airisuzuki_official_uf/1 >> shows invalid route

**Comment by Japhys** [comment]
Indeed, is not solved yet. Having the same issues

**Pull request merged**